packages/clerk-js/src/core/clerk.ts

```diff
@@ -873,6 +876,23 @@ export default class Clerk implements ClerkInterface {
         buildURL({ base: displayConfig.signUpUrl, hashPath: '/continue' }, { stringify: true }),
     );
 
+    const navigateToNextStepSignUp = () => {
+      if (su.unverifiedFields.length === 0 || su.missingFields.length !== 0) {
+        return navigateToContinueSignUp();
+      }
+
+      return handleMissingRequirements({
+        signUp,
+        verifyEmailPath:
+          params.verifyEmailAddressUrl ||
+          buildURL({ base: displayConfig.signUpUrl, hashPath: '/verify-email-address' }, { stringify: true }),
+        verifyPhonePath:
+          params.verifyPhoneNumberUrl ||
+          buildURL({ base: displayConfig.signUpUrl, hashPath: '/verify-phone-number' }, { stringify: true }),
+        navigate,
+      });
+    };
```

Comment:
🔧 As discussed offline, let's reuse the `completeSignUpFlow` here and avoid introducing more utility functions.
Given that this is a change that will affect one of our core flows, let's keep the changeset as straight forward as possible, to make sure that FE folks can review it easier and spot any issues.
If a refactoring is required, we can always do it in a subsequent PR.

Reply:
This `handleMissingRequirements` is being extracted from `completeSignUpFlow` and used in this. Do you mean revert that and use it as it was?

Reply:
Yes, just reuse the existing `completeSignUpFlow`. I understand the motivation behind this change, but given how  critical this part of the code is, I would leave the refactoring as a separate piece of work.